test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -81,33 +82,38 @@ def test_create_duplicate_cname_record_fails
   end
 
   def test_remove_address_records_with_longest_zone_match
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:get_ipv4_address!).with('host.foo.bar.domain.local')
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:enum_records).with('bar.domain.local', 'host.foo.bar.domain.local',  'A').returns(['1.1.1.1','2.2.2.2'])
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:remove_specific_record_from_zone).with('bar.domain.local', 'host.foo.bar.domain.local', '1.1.1.1', 'A').returns(true)
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:remove_specific_record_from_zone).with('bar.domain.local', 'host.foo.bar.domain.local', '2.2.2.2', 'A').returns(true)
     assert_nil @server.remove_a_record('host.foo.bar.domain.local')
   end
 
   def test_remove_a_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:remove_record).with('host.example.com','A').returns(nil)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:get_ipv4_address!).with('host.example.com')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:do_remove).with('host.example.com','A').returns(nil)
```

Comment:
As an illustration for the above, this and the next three tests should be the same for all providers and can be pulled into a dedicated suite.